operator/pkg/reconciliation/reconcile_racks.go

```diff
@@ -877,66 +789,53 @@ func isNodeStuckAfterLosingReadiness(pod *corev1.Pod) bool {
 	return hasBeenXMinutesSinceReady(10, pod)
 }
 
-func (rc *ReconciliationContext) getCassMetadataEndpoints(podList *corev1.PodList) map[string]httphelper.EndpointState {
-	var endpoints = make(map[string]httphelper.EndpointState)
-	for _, pod := range podList.Items {
+func (rc *ReconciliationContext) getCassMetadataEndpoints() httphelper.CassMetadataEndpoints {
+	var metadata httphelper.CassMetadataEndpoints
+	for _, pod := range rc.clusterPods {
 		// Try to query the first ready pod we find.
 		// We won't get any endpoints back if no pods are ready yet.
-		if !isServerReady(&pod) {
+		if !isServerReady(pod) {
 			continue
 		}
 
-		result, _ := rc.NodeMgmtClient.CallMetadataEndpointsEndpoint(&pod)
-		if len(result.Entity) == 0 {
+		metadata, _ := rc.NodeMgmtClient.CallMetadataEndpointsEndpoint(pod)
+		if len(metadata.Entity) == 0 {
 			continue
 		}
-
-		for _, ep := range result.Entity {
-			endpoints[ep.GetRpcAddress()] = ep
-		}
 		break
 	}
 
-	return endpoints
+	return metadata
 }
 
-func (rc *ReconciliationContext) deleteStuckNodes(podList *corev1.PodList) (bool, error) {
+func (rc *ReconciliationContext) deleteStuckNodes() (bool, error) {
 	rc.ReqLogger.Info("reconcile_racks::deleteStuckNodes")
-	for _, pod := range podList.Items {
+	for _, pod := range rc.dcPods {
 		shouldDelete := false
 		reason := ""
-		if isNodeStuckAfterTerminating(&pod) {
+		if isNodeStuckAfterTerminating(pod) {
 			reason = "Pod got stuck after Cassandra container terminated"
 			shouldDelete = true
-		} else if isNodeStuckAfterLosingReadiness(&pod) {
+		} else if isNodeStuckAfterLosingReadiness(pod) {
 			reason = "Pod got stuck after losing readiness"
 			shouldDelete = true
 		}
 
 		if shouldDelete {
 			rc.ReqLogger.Info(fmt.Sprintf("Deleting stuck pod: %s. Reason: %s", pod.Name, reason))
-			return true, rc.Client.Delete(rc.Ctx, &pod)
+			return true, rc.Client.Delete(rc.Ctx, pod)
 		}
 	}
 
 	return false, nil
 }
 
-func isClusterHealthy(rc *ReconciliationContext) bool {
-	selector := map[string]string{
-		api.ClusterLabel: rc.Datacenter.Spec.ClusterName,
-		// FIXME make a enum, pods should start in an Init state
-		api.CassNodeState: stateStarted,
-	}
-	podList, err := listPods(rc, selector)
-	if err != nil {
-		rc.ReqLogger.Error(err, "no started pods found for CassandraDatacenter")
-		return false
-	}
+func (rc *ReconciliationContext) isClusterHealthy() bool {
+	pods := FilterPodListByCassNodeState(rc.clusterPods, stateStarted)
```

Comment:
is it conventional for us to use "stateStarted" instead of STATE_STARTED ?

Reply:
Camel case seems to be the golang convention for constants. If we already are using full caps in other places though, I can change it.

Reply:
all caps is not a golang thing really, I spent time getting rid of that some months ago